src/Avalonia.Controls/Primitives/Popup.cs

```diff
@@ -326,7 +330,11 @@ public double HorizontalOffset
         public double VerticalOffset
         {
             get { return GetValue(VerticalOffsetProperty); }
-            set { SetValue(VerticalOffsetProperty, value); }
+            set
+            {
+                SetValue(VerticalOffsetProperty, value);
+                HandlePositionChange();
```

Comment:
Property changed handling shouldn't be in the property setter.
Use "override OnPropertyChanged" or static "VerticalOffsetProperty.Changed".

Reply:
This line will be executed only on setter call. But it also possible to change the value directly with SetValue(VerticalOffsetProperty, 10.0);

Reply:
@maxkatz6 Thanks, changed it